src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java

```diff
@@ -0,0 +1,73 @@
+package com.mercuryCyclists.Sales.controller;
+
+import com.mercuryCyclists.Sales.entity.InStoreSale;
+import com.mercuryCyclists.Sales.service.InStoreSaleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@CrossOrigin
+@RestController
+@RequestMapping(path = "api/v1/in-store-sale")
+public class InStoreSaleController {
+
+    private final InStoreSaleService inStoreSaleService;
+
+    @Autowired
+    public InStoreSaleController(InStoreSaleService inStoreSaleService) {
+        this.inStoreSaleService = inStoreSaleService;
+    }
+
+    /**
+     * Gets all In Store Sales
+     */
+    @GetMapping
+    public List<InStoreSale> getInStoreSales() {
+        return inStoreSaleService.getAllInStoreSales();
+    }
+
+    /**
+     * Gets In Store Sales by Store ID
+     */
+    @GetMapping(path = "/store/{storeId}/sales")
+    public List<InStoreSale> getStoreSales(@PathVariable("storeId") Long storeId) {
+        return inStoreSaleService.getStoreSales(storeId);
+    }
+
+    /**
+     * Register a new In Store Sale
+     */
+    @PostMapping(path = "/store/{storeId}")
+    public ResponseEntity<InStoreSale> registerInStoreSale(@RequestBody InStoreSale inStoreSale,
+                                                           @PathVariable("storeId") Long storeId) {
+        return inStoreSaleService.registerInStoreSale(inStoreSale, storeId);
+    }
+
+    /**
+     * Add store to in store sale
+     */
+    @PostMapping(path = "/{inStoreSaleId}/store/{storeId}/add-store")
+    public InStoreSale addStoreToInStoreSale(@PathVariable("inStoreSaleId") Long inStoreSaleId,
+                                             @PathVariable("storeId") Long storeId) {
+        return inStoreSaleService.addStoreToInStoreSale(inStoreSaleId, storeId);
+    }
+
+    /**
+     * Updates existing in store sale based on the in store sale given
+     */
+    @PutMapping(path = "/{inStoreSaleId}")
+    public InStoreSale updateInStoreSale(@RequestBody InStoreSale inStoreSale,
+                                         @PathVariable("inStoreSaleId") Long inStoreSaleId) {
+        return inStoreSaleService.updateInStoreSale(inStoreSale, inStoreSaleId);
+    }
```

Comment:
There's probably a good argument somewhere that we shouldnt be updating pre-existing sales as it'll cause a shitshow. I don't mind either way; but we gotta make sure that we don't update the quanities